docs/topics/db/aggregation.txt

```diff
@@ -632,3 +632,46 @@ aggregate that author count, referencing the annotation field:
     >>> from django.db.models import Avg, Count
     >>> Book.objects.annotate(num_authors=Count("authors")).aggregate(Avg("num_authors"))
     {'num_authors__avg': 1.66}
+
+Handling empty querysets using aggregate ``default`` 
+----------------------------------------------------
```

Comment:
It might be worth explaining that aggregate functions are simply returning what the database is returning. (I'm pretty sure that is the case someone may correct me.)  This pretty much explains all of the behaviour including counts returning 0.